tensorflow_addons/metrics/f1_scores.py

```diff
@@ -206,3 +248,103 @@ def reset_states(self):
+
+    F1 score is the weighted average of precision and
+    recall. Output range is [0, 1]. This works for both
+    multi-class and multi-label classification.
+
+    F-1 = (2) * ((precision * recall) / (precision + recall))
+
+    Args:
+       num_classes : Number of unique classes in the dataset.
+       average : Type of averaging to be performed on data.
+                 Acceptable values are `None`, `micro`, `macro` and
+                 `weighted`.
+                 Default value is None.
+       beta : float
+              Determines the weight of precision and recall in harmonic
+              mean. It's value is 1.0 for F1 score.
+
+    Returns:
+       F1 Score: float
+
+    Raises:
+       ValueError: If the `average` has values other than
+       [None, micro, macro, weighted].
+
+       ValueError: If the `beta` value is less than or equal
+       to 0.
+
+    `average` parameter behavior:
+
+    1. If `None` is specified as an input, scores for each
+       class are returned.
+
+    2. If `micro` is specified, metrics like true positivies,
+       false positives and false negatives are computed
+       globally.
+
+    3. If `macro` is specified, metrics like true positivies,
+       false positives and false negatives are computed for
+       each class and their unweighted mean is returned.
+       Imbalance in dataset is not taken into account for
+       calculating the score
+
+    4. If `weighted` is specified, metrics are computed for
+       each class and returns the mean weighted by the
+       number of true instances in each class taking data
+       imbalance into account.
+
+    Usage:
+
+    F1 Micro Score
+
+    ```python
+    # F1 Micro
+    actuals = tf.constant([[1, 1, 0],[1, 0, 0]],
+              dtype=tf.int32)
+    preds = tf.constant([[1, 0, 0],[1, 0, 1]],
+              dtype=tf.int32)
+    output = tf.keras.metrics.F1Score(num_classes=3,
+              average='micro')
+    output.update_state(actuals, preds)
+    print('F1 Micro score is: ',
+            output.result().numpy()) # 0.6666667
+    # F1 Macro
+    output = tf.keras.metrics.F1Score(num_classes=3,
+                average='macro')
+    output.update_state(actuals, preds)
+    print('F1 Macro score is: ',
+            output.result().numpy()) # 0.33333334
+    # F1 weighted
+    output = tf.keras.metrics.F1Score(num_classes=3,
+              average='weighted')
+    output.update_state(actuals, preds)
+    print('F1 Weighted score is: ',
+            output.result().numpy()) # 0.6666667
+    # F1 score for each class (average=None).
+    output = tf.keras.metrics.F1Score(num_classes=3)
+    output.update_state(actuals, preds)
+    print('F1 score is: ',
+            output.result().numpy()) # [1. 0. 0.]
+    ```
+    """
+
+    def __init__(self, num_classes, average, name='f1_score',
+                 dtype=tf.float32):
+        super().__init__(num_classes, average, 1.0, name=name, dtype=dtype)
```

Comment:
lets make the style unified in the same file though this is also correct. Use
```python
super(F1Score, self).__init__(...)
```
instead.